### ISSUE TITLE
check frequency of gradient updates in the original ALCOVE model

### ISSUE THREAD

**Issue by wkvong** [alexatartaglini/alcove#9]
does the gradient update in the original ALCOVE paper occur after each example, or after the whole set of examples? right now, our version of the code performs a gradient update after passing in all of the examples, but we should modify this if necessary

**Comment by alexatartaglini** [comment]
<img width="360" alt="Screen Shot 2020-04-09 at 10 07 04 AM" src="https://user-images.githubusercontent.com/48462071/78915926-ea47d980-7a49-11ea-81b8-8f9ab3531c53.png">
On pg. 24, the paper indicates that it updates after each exemplar is passed in.

**Comment by wkvong** [comment]
Cool, good catch Alexa! I'd say the two things to try out to match the original paper then is to:

(1) For the loss function, try it out using reduction='sum' rather than reduction='mean', this will cause larger updates since we are summing over the loss from the all examples and may produce a closer match to the learning curves (but still only performing the update after seeing all examples, so is not technically the same)
(2) Modify the pytorch code to perform an update after each example rather than after all of the examples
